sign/pending_requests.go

```diff
@@ -114,7 +114,11 @@ func (rs *PendingRequests) Wait(id string, timeout time.Duration) Result {
 		case rst := <-request.result:
 			return rst
 		case <-time.After(timeout):
-			rs.complete(request, EmptyResponse, ErrSignReqTimedOut)
+			_, err := rs.tryLock(request.ID)
+			// if request is not already in progress, we complete it.
+			if err == nil {
```

Comment:
what if error is inprogress?

Reply:
`tryLock` can return errors if
a) there is no such request
b) `Approve` is already called and in-progress now
c) another timeout is in progress now

for me, it makes sense, that we only proceed with a timeout if `Approve` wasn't called before the timeout fires

Reply:
> b) Approve is already called and in-progress now

so for example, if i approve transaction and it makes a network request which takes longer than timeout passed to wait and eventually times out - only caller of the Approve will receive an error?

this might be a problem if caller of Approve is user and caller of the Wait is dapp, or workflow is different?

Reply:
Nope, because that is an error of a `completeFunc` and it gets propagated to `Wait` too.
But yeah, if the `completeFunc` never ever finishes, `Wait` will lock.

If we don't do that, we will end up in a really weird place, where a request can both succeed (the call gets propagated later) and timeout at the same time.

Reply:
thanks, now i recall what was the reason for for { select {}} in the wait
seems good to assume that once we are in Approve we need to wait until approve will finish